src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6474,6 +6474,13 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
                                 Debug.Assert(!leftType.IsDynamic());
                                 Debug.Assert(IsPotentialColorColorReceiver(left, leftType));
 
+                                if (leftSymbol is SourceFieldSymbolWithSyntaxReference fieldLeft)
+                                {
+                                    boundValue = RebindColorColorConstField(left, boundValue, fieldLeft);
+                                    leftSymbol = boundValue.ExpressionSymbol;
```

Comment:
>leftSymbol = boundValue.ExpressionSymbol; [](http://example.com/codeflow?start=36&length=41)

Why is this necessary? Do we expect the symbol to change?

Reply:
We do rebind the symbol, and upon returning the bound expression a few lines below this, we also provide the left symbol that is the dependency for this constant's dependency graph, resolving the issue with the circular reference. After having successfully rebound the symbol, we avoid introducing the field itself into the dependency graph as well.

Reply:
> We do rebind the symbol, and upon returning the bound expression a few lines below this, we also provide the left symbol that is the dependency for this constant's dependency graph, resolving the issue with the circular reference. After having successfully rebound the symbol, we avoid introducing the field itself into the dependency graph as well.

This doesn't answer the question: "Do we expect the symbol to change?"

Reply:
More specifically. Why is this the right thing to replace the field symbol with a symbol that the enclosing member access resolves to?